addons/survey/tests/test_survey.py

```diff
@@ -475,3 +477,57 @@ def test_get_pages_and_questions_to_show(self):
         returned_questions_and_pages = my_survey._get_pages_and_questions_to_show()
 
         self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
+
+    @users('survey_manager')
+    def test_allowed_triggering_question_ids(self):
+        # Create 2 surveys, each with 3 questions, each with 2 suggested answers
+        survey_1, survey_2 = self.env['survey.survey'].create([
+            {'title': 'Test Survey 1'},
+            {'title': 'Test Survey 2'}
+        ])
+        self.env['survey.question'].create([
+            {
+                'survey_id': survey.id,
+                'title': f'Question {jdx}',
+                'question_type': 'simple_choice',
+                'suggested_answer_ids': [
+                    Command.create({
+                        'value': f'Answer {kdx}',
+                    }) for kdx in range(2)],
+            }
+            for jdx in range(3)
+            for survey in (survey_1, survey_2)
+        ])
+        survey_1_q_1, survey_1_q_2, _ = survey_1.question_ids
+        survey_2_q_1, survey_2_q_2, _ = survey_2.question_ids
+
+        # Make sure they are already configured as conditionally triggered
+        survey_1_q_2.write({
+            'is_conditional': True,
+            'triggering_question_id': survey_1_q_1,
+            'triggering_answer_id': survey_1_q_1.suggested_answer_ids[0]
+        })
+        survey_2_q_2.write({
+            'is_conditional': True,
+            'triggering_question_id': survey_2_q_1,
+            'triggering_answer_id': survey_2_q_1.suggested_answer_ids[0]
+        })
+
+        with self.subTest("Editing existing questions"):
+            # Only previous questions from the same survey
+            self.assertFalse(survey_1_q_2.allowed_triggering_question_ids & survey_2_q_2.allowed_triggering_question_ids)
+            self.assertEqual(survey_1_q_2.allowed_triggering_question_ids, survey_1_q_1)
+            self.assertEqual(survey_2_q_2.allowed_triggering_question_ids, survey_2_q_1)
+
+        (survey_1_q_2 | survey_2_q_2).invalidate_recordset()
+
+        class NoId:
+            def __init__(self):
+                self.id = False
```

Comment:
Is that something you picked from somewhere else?

Can't you use self.env['survey.question'].new() (instead of create) to better test that?

Would be something like:
`survey_1_q_2 = self.env['survey.question'].new(survey_1_q_2.copy_data())`

Never tried this myself so maybe it does not work.
But redefining a NoId class looks a bit strange (and may be too dependent on low-level ORM implementation).

Reply:
I simply didn't know `new()`... which is much better indeed. Thanks!